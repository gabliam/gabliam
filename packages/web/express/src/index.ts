import { ExpressPlugin } from './express-plugin';

export * from 'http-status-codes';
export * from './express';
export * from './express-converter';
export * from './express-interceptor';
export * from './interfaces';
export * from './request';
export * from './response';
export { ExpressPlugin as default };
