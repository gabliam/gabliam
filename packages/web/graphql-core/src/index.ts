export { TYPE } from './constants';
export * from './graphql-core-plugin';
export * from './interfaces';
export * from './resolver-decorator';
export {
  Arg,
  ArgOptions,
  Args,
  ArgsType,
  Authorized,
  createParamDecorator,
  createMethodDecorator,
  Ctx,
  Directive,
  Extensions,
  registerEnumType,
  Field,
  FieldOptions,
  FieldResolver,
  Info,
  InputType,
  InputTypeOptions,
  InterfaceType,
  InterfaceTypeOptions,
  Mutation,
  ObjectType,
  ObjectTypeOptions,
  PubSub,
  Query,
  Resolver,
  Root,
  Subscription,
  SubscriptionOptions,
  createUnionType,
  UseMiddleware,
  ID, Int, ClassType, Publisher
} from 'type-graphql';
