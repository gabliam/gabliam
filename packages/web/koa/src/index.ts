import { KoaPlugin } from './koa-plugin';

export * from 'http-status-codes';
export * from './interfaces';
export * from './koa';
export * from './koa-converter';
export * from './koa-interceptor';
export * from './request';
export * from './response';
export { KoaPlugin as default };
