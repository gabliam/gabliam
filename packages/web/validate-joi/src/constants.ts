export const METADATA_KEY = {
  validate: '_validate',
};

export const ERRORS_MSGS = {
  DUPLICATED_VALIDATE_DECORATOR: `Cannot apply @Validate decorator multiple times.`,
};
