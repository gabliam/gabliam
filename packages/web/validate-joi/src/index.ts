export * from './metadatas';
export * from './validate';
export * from './constants';
export * from './validate-config';
export { ValidatePlugin as default } from './validate-plugin';
