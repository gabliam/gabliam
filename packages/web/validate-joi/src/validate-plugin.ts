import { Plugin, Scan } from '@gabliam/core';

@Plugin()
@Scan()
export class ValidatePlugin {}
