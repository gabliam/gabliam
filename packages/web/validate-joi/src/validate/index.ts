export * from './validate-interceptor';
export * from './validate-request';
export * from './validate-senderror-interceptor';
