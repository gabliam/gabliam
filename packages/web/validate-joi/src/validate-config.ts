import { Config, Init } from '@gabliam/core';
import { WebConfiguration } from '@gabliam/web-core';
import { ValidateSendErrorInterceptor, ValidateInterceptor } from './validate';

@Config()
export class ValidateConfig {
  constructor(private webConfiguration: WebConfiguration) {}

  @Init()
  init() {
    this.webConfiguration.useGlobalInterceptor(
      ValidateSendErrorInterceptor,
      ValidateInterceptor
    );
  }
}
