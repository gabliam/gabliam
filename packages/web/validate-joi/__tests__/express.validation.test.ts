import { Joi } from '@gabliam/core';
import {
  Controller,
  Get,
  Post,
  QueryParam,
  RequestBody,
  RequestHeaders,
  RequestParam,
} from '@gabliam/web-core';
import * as supertest from 'supertest';
import { Validate } from '../src';
import { ExpressPluginTest } from './express-plugin-test';

let appTest: ExpressPluginTest;

beforeEach(async () => {
  appTest = new ExpressPluginTest();
});

afterEach(async () => {
  await appTest.destroy();
});

describe('express integration', () => {
  describe('validations', () => {
    it('req.headers', async () => {
      @Controller('/')
      class TestController {
        @Get('/')
        @Validate(
          {
            headers: {
              accept: Joi.string().regex(/xml/),
            },
          },
          { allowUnknown: true }
        )
        async getTest() {
          return 'lol';
        }
      }

      appTest.addClass(TestController);
      await appTest.build();
      const response = await supertest(appTest.app)
        .get('/')
        .set('Accept', 'application/json')
        .expect(400);
      expect(response).toMatchSnapshot();
    });

    it('req.params', async () => {
      @Controller('/')
      class TestController {
        @Get('/user/:id')
        @Validate({
          params: {
            id: Joi.string().token(),
          },
        })
        async getTest() {
          return 'lol';
        }
      }

      appTest.addClass(TestController);
      await appTest.build();
      const response = await supertest(appTest.app)
        .get('/user/@@')
        .expect(400);
      expect(response).toMatchSnapshot();
    });

    it('req.query', async () => {
      @Controller('/')
      class TestController {
        @Get('/')
        @Validate({
          query: Joi.object().keys({
            start: Joi.date(),
          }),
        })
        async getTest() {
          return 'lol';
        }
      }

      appTest.addClass(TestController);
      await appTest.build();
      const response = await supertest(appTest.app)
        .get('/?end=lol')
        .expect(400);
      expect(response).toMatchSnapshot();
    });

    it('req.body', async () => {
      @Controller('/')
      class TestController {
        @Post('/')
        @Validate({
          body: {
            first: Joi.string().required(),
            last: Joi.string(),
            role: Joi.number().integer(),
          },
        })
        async getTest() {
          return 'lol';
        }
      }

      appTest.addClass(TestController);

      await appTest.build();
      const response = await supertest(appTest.app)
        .post('/')
        .send({
          first: 'john',
          last: 123,
        })
        .expect(400);
      expect(response).toMatchSnapshot();
    });
  });

  describe('update req values', () => {
    it('req.headers', async () => {
      @Controller('/')
      class TestController {
        @Get('/')
        @Validate(
          {
            headers: {
              accept: Joi.string().regex(/json/),
              id: Joi.number().default(12),
              'secret-header': Joi.string().default('@@@@@@'),
            },
          },
          { allowUnknown: true }
        )
        async getTest(@RequestHeaders() headers: { [k: string]: any }) {
          delete headers.host; // this can change computer to computer, so just remove it
          expect(headers).toMatchSnapshot();
          return 'k';
        }
      }

      appTest.addClass(TestController);
      await appTest.build();
      const response = await supertest(appTest.app)
        .get('/')
        .set('Accept', 'application/json')
        .expect(200);
      expect(response).toMatchSnapshot();
    });

    it('req.params', async () => {
      @Controller('/')
      class TestController {
        @Get('/:id/user/:name')
        @Validate({
          params: {
            id: Joi.number(),
            name: Joi.string().uppercase(),
          },
        })
        async getTest(
          @RequestParam('id') id: number,
          @RequestParam('name') name: string
        ) {
          expect(id).toBe(12);
          expect(name).toBe('ADAM');
          return 'k';
        }
      }

      appTest.addClass(TestController);
      await appTest.build();
      const response = await supertest(appTest.app)
        .get('/12/user/adam')
        .expect(200);
      expect(response).toMatchSnapshot();
    });

    it('req.query', async () => {
      @Controller('/')
      class TestController {
        @Get('/')
        @Validate({
          query: Joi.object().keys({
            name: Joi.string().uppercase(),
            page: Joi.number().default(1),
            id: Joi.number().required(),
          }),
        })
        async getTest(@QueryParam() query: any) {
          expect(query).toEqual({
            name: 'JOHN',
            page: 1,
            id: 42,
          });
          return 'k';
        }
      }

      appTest.addClass(TestController);
      await appTest.build();
      const response = await supertest(appTest.app)
        .get('/?name=john&id=42')
        .expect(200);
      expect(response).toMatchSnapshot();
    });

    it('req.body', async () => {
      @Controller('/')
      class TestController {
        @Post('/')
        @Validate({
          body: {
            first: Joi.string().required(),
            last: Joi.string().default('Smith'),
            role: Joi.string().uppercase(),
            id: Joi.number().default(42),
          },
        })
        async getTest(@RequestBody() body: any) {
          expect(body).toEqual({
            first: 'john',
            role: 'ADMIN',
            last: 'Smith',
            id: 42,
          });
          return 'lol';
        }
      }

      appTest.addClass(TestController);

      await appTest.build();
      const response = await supertest(appTest.app)
        .post('/?end=lol')
        .send({
          first: 'john',
          role: 'admin',
        })
        .expect(200);
      expect(response).toMatchSnapshot();
    });
  });
});
