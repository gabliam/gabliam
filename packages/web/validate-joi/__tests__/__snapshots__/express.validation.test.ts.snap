// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`express integration update req values req.body 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "3",
    "content-type": "text/html; charset=utf-8",
  },
  "req": Object {
    "data": Object {
      "first": "john",
      "role": "admin",
    },
    "headers": Object {
      "content-type": "application/json",
    },
    "method": "POST",
  },
  "status": 200,
  "text": "lol",
}
`;

exports[`express integration update req values req.headers 1`] = `
Object {
  "accept": "application/json",
  "accept-encoding": "gzip, deflate",
  "connection": "close",
  "id": 12,
  "secret-header": "@@@@@@",
  "user-agent": "node-superagent/3.8.3",
}
`;

exports[`express integration update req values req.headers 2`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "1",
    "content-type": "text/html; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {
      "accept": "application/json",
    },
    "method": "GET",
  },
  "status": 200,
  "text": "k",
}
`;

exports[`express integration update req values req.params 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "1",
    "content-type": "text/html; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "GET",
  },
  "status": 200,
  "text": "k",
}
`;

exports[`express integration update req values req.query 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "1",
    "content-type": "text/html; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "GET",
  },
  "status": 200,
  "text": "k",
}
`;

exports[`express integration validations req.body 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "156",
    "content-type": "application/json; charset=utf-8",
  },
  "req": Object {
    "data": Object {
      "first": "john",
      "last": 123,
    },
    "headers": Object {
      "content-type": "application/json",
    },
    "method": "POST",
  },
  "status": 400,
  "text": "{\\"statusCode\\":400,\\"error\\":\\"Bad Request\\",\\"message\\":\\"child /\\"last/\\" fails because [/\\"last/\\" must be a string]\\",\\"validation\\":{\\"source\\":\\"body\\",\\"keys\\":[\\"last\\"]}}",
}
`;

exports[`express integration validations req.headers 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "228",
    "content-type": "application/json; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {
      "accept": "application/json",
    },
    "method": "GET",
  },
  "status": 400,
  "text": "{\\"statusCode\\":400,\\"error\\":\\"Bad Request\\",\\"message\\":\\"child /\\"accept/\\" fails because [/\\"accept/\\" with value /\\"application&#x2f;json/\\" fails to match the required pattern: /xml/]\\",\\"validation\\":{\\"source\\":\\"headers\\",\\"keys\\":[\\"accept\\"]}}",
}
`;

exports[`express integration validations req.params 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "193",
    "content-type": "application/json; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "GET",
  },
  "status": 400,
  "text": "{\\"statusCode\\":400,\\"error\\":\\"Bad Request\\",\\"message\\":\\"child /\\"id/\\" fails because [/\\"id/\\" must only contain alpha-numeric and underscore characters]\\",\\"validation\\":{\\"source\\":\\"params\\",\\"keys\\":[\\"id\\"]}}",
}
`;

exports[`express integration validations req.query 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "122",
    "content-type": "application/json; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "GET",
  },
  "status": 400,
  "text": "{\\"statusCode\\":400,\\"error\\":\\"Bad Request\\",\\"message\\":\\"/\\"end/\\" is not allowed\\",\\"validation\\":{\\"source\\":\\"query\\",\\"keys\\":[\\"end\\"]}}",
}
`;
